File: src/backend/app/routes/material.py
"""
材料库API路由
"""
from flask import Blueprint, jsonify

material_bp = Blueprint('material', __name__, url_prefix='')


@material_bp.route('/materials', methods=['GET'])
def get_materials():
    """
    获取常用PCB材料的介电常数
    GET /api/materials
    """
    materials = {
        'FR4': {'er': 4.3, 'loss_tangent': 0.02, 'name': 'FR4 (标准)'},
        'FR4_HF': {'er': 4.1, 'loss_tangent': 0.015, 'name': 'FR4 (高频)'},
        'Rogers4003C': {'er': 3.38, 'loss_tangent': 0.0027, 'name': 'Rogers 4003C'},
        'Rogers4350B': {'er': 3.48, 'loss_tangent': 0.0037, 'name': 'Rogers 4350B'},
        'Isola370HR': {'er': 4.04, 'loss_tangent': 0.019, 'name': 'Isola 370HR'},
        'Teflon': {'er': 2.1, 'loss_tangent': 0.0002, 'name': 'Teflon/PTFE'},
        'Polyimide': {'er': 3.4, 'loss_tangent': 0.008, 'name': 'Polyimide'},
    }
    return jsonify(materials), 200

File: src/backend/api/material.js
/**
 * 材料库API处理函数
 * 移植自 backend/app/routes/material.py
 */

const materialsDatabase = {
  'FR4': {
    name: 'FR4',
    dielectric: 4.5,
    tanD: 0.015,
    frequency: '1 MHz',
    description: '标准玻璃纤维增强环氧树脂',
    frequency_range: '1MHz - 5GHz'
  },
  'FR4_HF': {
    name: 'FR4 (High Frequency)',
    dielectric: 4.2,
    tanD: 0.008,
    frequency: '1 GHz',
    description: '高频FR4，损耗更低',
    frequency_range: '1GHz - 20GHz'
  },
  'Rogers_4003C': {
    name: 'Rogers 4003C',
    dielectric: 3.55,
    tanD: 0.0027,
    frequency: '10 GHz',
    description: '微波级陶瓷填充聚四氟乙烯',
    frequency_range: '1GHz - 50GHz'
  },
  'Rogers_4350B': {
    name: 'Rogers 4350B',
    dielectric: 3.48,
    tanD: 0.004,
    frequency: '10 GHz',
    description: '高性能微波PCB材料',
    frequency_range: '1GHz - 40GHz'
  },
  'Isola_370HR': {
    name: 'Isola 370HR',
    dielectric: 3.66,
    tanD: 0.009,
    frequency: '10 GHz',
    description: '低损耗高可靠性PCB材料',
    frequency_range: '1GHz - 30GHz'
  },
  'Teflon_PTFE': {
    name: 'Teflon / PTFE',
    dielectric: 2.1,
    tanD: 0.0005,
    frequency: '10 GHz',
    description: '聚四氟乙烯，极低损耗',
    frequency_range: '1GHz - 100GHz'
  },
  'Polyimide': {
    name: 'Polyimide',
    dielectric: 3.5,
    tanD: 0.01,
    frequency: '1 GHz',
    description: '聚酰亚胺，高温稳定性',
    frequency_range: '1MHz - 10GHz'
  },
  'Ceramic': {
    name: 'Ceramic',
    dielectric: 9.8,
    tanD: 0.002,
    frequency: '10 GHz',
    description: '陶瓷基复合材料',
    frequency_range: '1GHz - 50GHz'
  }
}

export const handleMaterial = async (request) => {
  try {
    const url = new URL(request.url)
    const pathParts = url.pathname.split('/')
    const name = pathParts[pathParts.length - 1]

    if (name === 'materials') {
      // 返回所有材料
      return new Response(
        JSON.stringify({
          materials: Object.values(materialsDatabase),
          count: Object.keys(materialsDatabase).length
        }),
        { headers: { 'Content-Type': 'application/json' } }
      )
    }

    if (materialsDatabase[name]) {
      return new Response(
        JSON.stringify(materialsDatabase[name]),
        { headers: { 'Content-Type': 'application/json' } }
      )
    }

    return new Response(JSON.stringify({ error: 'Material not found' }), {
      status: 404,
      headers: { 'Content-Type': 'application/json' }
    })
  } catch (error) {
    return new Response(JSON.stringify({ error: error.message }), {
      status: 500,
      headers: { 'Content-Type': 'application/json' }
    })
  }
}
